src/NServiceBus.Gateway/Routing/RouteToGatewayBehaviour.cs

```diff
@@ -1,23 +1,44 @@
 ﻿namespace NServiceBus
 {
     using System;
+    using System.Collections.Generic;
     using System.Threading.Tasks;
     using Gateway.Routing;
     using Pipeline;
     using Routing;
 
     class RouteToGatewayBehavior : Behavior<IRoutingContext>
     {
-        public RouteToGatewayBehavior(string gatewayAddress)
+        public RouteToGatewayBehavior(string gatewayAddress, IList<string> configuredSiteKeys)
         {
             this.gatewayAddress = gatewayAddress;
+            this.configuredSiteKeys = configuredSiteKeys;
         }
 
         public override Task Invoke(IRoutingContext context, Func<Task> next)
         {
-
             if (context.Extensions.TryGet(out RouteThroughGateway _))
             {
+                if (context.Message.Headers.TryGetValue(Headers.DestinationSites, out var siteKeyHeader))
+                {
+                    var siteKeys = siteKeyHeader.Split(',');
```

Comment:
It makes me sad that we take these sites in as a collection, join them to save them, and then split them back out again. Seems like we could do better here and just save them as a collection?

Reply:
Talking to @timbussmann we both agree that this is a valid change but not in scope for the patch since its more of a refactoring. I will raise an issue to track this and flag it as a candidate for a maintenance release

Reply:
Raise https://github.com/Particular/NServiceBus.Gateway/issues/106